### PR TITLE
Configure vm metadata along with creating object

### PR DESCRIPTION
**Pull request by jianqiu**
Using `userData` in `SoftLayer_Virtual_Guest_Template` to attach vm user data in the process of creating object,  `ConfigureMetadataOnVirtualGuest` after creating object is not needed which could save time.

About `ephemeral` disk,  keep consistent with legacy softlayer cpi, make creating ephemeral disk as optional which is indicated by `ephemeralDiskSize` in `VMCloudProperties`. If `ephemeralDiskSize ==0`, softlayer_creator would not create and attach `ephemeral` disk,  just wait for the vm power on. what follows is the changes in `softlayer_creator.go` in softlayer vm package:
```go
	if cloudProps.EphemeralDiskSize == 0 {
		err = bslcommon.WaitForVirtualGuest(c.softLayerClient, virtualGuest.Id, "RUNNING", bslcommon.TIMEOUT, bslcommon.POLLING_INTERVAL)
		if err != nil {
			return SoftLayerVM{}, bosherr.WrapError(err, fmt.Sprintf("PowerOn failed with VirtualGuest id `%d`", virtualGuest.Id))
		}
	} else {
		err = bslcommon.AttachEphemeralDiskToVirtualGuest(c.softLayerClient, virtualGuest.Id, cloudProps.EphemeralDiskSize, bslcommon.TIMEOUT, bslcommon.POLLING_INTERVAL)
		if err != nil {
			return SoftLayerVM{}, bosherr.WrapError(err, fmt.Sprintf("Attaching ephemeral disk to VirtualGuest `%d`", virtualGuest.Id))
		}
	}
```

### CODE REVIEW

File: integration/has_vm/has_vm_test.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ var _ = Describe("BOSH Director Level Integration for has_vm", func() {
 
 			outputBytes, err := testhelperscpi.RunCpi(rootTemplatePath, tmpConfigPath, jsonPayload)
 			Expect(err).ToNot(HaveOccurred())
-			
+
```
Comment (maximilien) [inline integration/has_vm/has_vm_test.go]: Why new line?

File: integration/has_vm/has_vm_test.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ var _ = Describe("BOSH Director Level Integration for has_vm", func() {
...
 			err = json.Unmarshal(outputBytes, &output)
```
Comment (maximilien) [inline integration/has_vm/has_vm_test.go]: Why new line?

**Comment by maximilien** [comment]
Please address small comment and remove TODO either by moving it to a story or if no longer needed then remove. Thanks.

**Comment by jianqiu** [comment]
@maximilien 
Thanks, just updated. I will create specific stories in our pivotaltracker, just removed `TODO` comments from code.

**Comment by maximilien** [comment]
Looks good. Merging. Cheers :beers: 

PS: please ensure we have integration tests for this (direct or indirect) and add chore story if not. Also please monitor the new pipeline to ensure that this code passes integration tests.

**Pull request merged**